internal/pkg/addon/addons.go

```diff
@@ -68,54 +58,75 @@ func New(wlName string) (*Addons, error) {
 	}, nil
 }
 
-// NewPackager creates an Addons struct that will package local artifacts when
-// generating the addons template.
-// See https://docs.aws.amazon.com/cli/latest/reference/cloudformation/package.html for more details.
-func NewPackager(wlName string, bucket string, uploader uploader) (*Addons, error) {
-	addons, err := New(wlName)
+type Stack struct {
+	template   *cfnTemplate
+	parameters yaml.Node
+	wlName     string
+}
+
+func (a *Addons) Stack() (*Stack, error) {
```

Comment:
I wonder if we need the method 🤔 it seems like clients are always forced to do 2-steps:
```go
addons := addons.New(/* params... */)
stack, err := addons.Stack()
```
Can we get rid of the `New` constructor and `Stack` method and replace it with a single `Parse` function:
```go
// Parse reads the addons/ directory under the workloadName and returns a stack configuration.
func Parse(workloadName string, ws WorkspaceReader) (*Stack, error) {
  // New + addons.Stack()
}
```

Reply:
i like this idea! def should help clean it up